docs/rfcs/751-write-empty-file-behavior.md

```diff
@@ -0,0 +1,62 @@
+- Author:  Abyss-w <mad.hatter@foxmail.com>
+- Start Date: 2021-09-10
+- RFC PR: [beyondstorage/go-storage#751](https://github.com/beyondstorage/go-storage/pull/751)
+- Tracking Issue: [beyondstorage/go-storage#0](https://github.com/beyondstorage/go-storage/issues/0)
+
+# GSP-751: Write Empty File Behavior
+
+- Previous discussion:
+  - [Specify the behavior for writing empty file](https://forum.beyondstorage.io/t/topic/204)
+
+## Background
+
+```go
+func (s *Storage) Write(path string, r io.Reader, size int64, pairs ...Pair) (n int64, err error)
+```
+
+In our definition, the `Write` API will upload a file to the path. We do not do anything about uploading empty files(i.e. `io.Reader`is nil ). However, some services do not support uploading nil `io.Reader`. As [proposed](https://forum.beyondstorage.io/t/topic/204) in our community, for s3, when we upload an empty file, the service will hang permanently. Here is an example of s3:
+
+```go
+_, err = s3.s.Write(path,nil, 0)

```

Comment:
Please format the code correctly.

Reply:
Maybe we can use `_, err = store.Write(path, nil, 0)` here so that our readers don't need to care about what `s3.s` is.